tests/test_wasm_builtin_simd.c

```diff
@@ -0,0 +1,488 @@
+i16x8 TESTFN i16x8_gt_s(i16x8 x, i16x8 y) {
+  return x > y;
+}
+i16x8 TESTFN i16x8_gt_u(i16x8 x, i16x8 y) {
+  return (u16x8)x > (u16x8)y;
+}
+i16x8 TESTFN i16x8_le_s(i16x8 x, i16x8 y) {
+  return x <= y;
+}
+i16x8 TESTFN i16x8_le_u(i16x8 x, i16x8 y) {
+  return (u16x8)x <= (u16x8)y;
+}
+i16x8 TESTFN i16x8_ge_s(i16x8 x, i16x8 y) {
+  return x >= y;
+}
+i16x8  TESTFN i16x8_ge_u(i16x8 x, i16x8 y) {
+  return (u16x8)x >= (u16x8)y;
+}
+i32x4 TESTFN i32x4_eq(i32x4 x, i32x4 y) {
+  return (i32x4)(x == y);
+}
+i32x4 TESTFN i32x4_ne(i32x4 x, i32x4 y) {
+  return (i32x4)(x != y);
+}
+i32x4 TESTFN i32x4_lt_s(i32x4 x, i32x4 y) {
+  return (i32x4)(x < y);
+}
+i32x4 TESTFN i32x4_lt_u(i32x4 x, i32x4 y) {
+  return (i32x4)((u32x4)x < (u32x4)y);
+}
+i32x4 TESTFN i32x4_gt_s(i32x4 x, i32x4 y) {
+  return (i32x4)(x > y);
+}
+i32x4 TESTFN i32x4_gt_u(i32x4 x, i32x4 y) {
+  return (i32x4)((u32x4)x > (u32x4)y);
+}
+i32x4 TESTFN i32x4_le_s(i32x4 x, i32x4 y) {
+  return (i32x4)(x <= y);
+}
+i32x4 TESTFN i32x4_le_u(i32x4 x, i32x4 y) {
+  return (i32x4)((u32x4)x <= (u32x4)y);
+}
+i32x4 TESTFN i32x4_ge_s(i32x4 x, i32x4 y) {
+  return (i32x4)(x >= y);
+}
+i32x4  TESTFN i32x4_ge_u(i32x4 x, i32x4 y) {
+  return (i32x4)((u32x4)x >= (u32x4)y);
+}
+i32x4 TESTFN f32x4_eq(f32x4 x, f32x4 y) {
+  return (i32x4)(x == y);
+}
+i32x4 TESTFN f32x4_ne(f32x4 x, f32x4 y) {
+  return (i32x4)(x != y);
+}
+i32x4 TESTFN f32x4_lt(f32x4 x, f32x4 y) {
+  return (i32x4)(x < y);
+}
+i32x4 TESTFN f32x4_gt(f32x4 x, f32x4 y) {
+  return (i32x4)(x > y);
+}
+i32x4 TESTFN f32x4_le(f32x4 x, f32x4 y) {
+  return (i32x4)(x <= y);
+}
+i32x4 TESTFN f32x4_ge(f32x4 x, f32x4 y) {
+  return (i32x4)(x >= y);
+}
+i64x2 TESTFN f64x2_eq(f64x2 x, f64x2 y) {
+  return (i64x2)(x == y);
+}
+i64x2 TESTFN f64x2_ne(f64x2 x, f64x2 y) {
+  return (i64x2)(x != y);
+}
+i64x2 TESTFN f64x2_lt(f64x2 x, f64x2 y) {
+  return (i64x2)(x < y);
+}
+i64x2 TESTFN f64x2_gt(f64x2 x, f64x2 y) {
+  return (i64x2)(x > y);
+}
+i64x2 TESTFN f64x2_le(f64x2 x, f64x2 y) {
+  return (i64x2)(x <= y);
+}
+i64x2 TESTFN f64x2_ge(f64x2 x, f64x2 y) {
+  return (i64x2)(x >= y);
+}
+v128 TESTFN v128_not(v128 vec) {
+  return ~vec;
+}
+v128 TESTFN v128_and(v128 x, v128 y) {
+  return x & y;
+}
+v128 TESTFN v128_or(v128 x, v128 y) {
+  return x | y;
+}
+v128 TESTFN v128_xor(v128 x, v128 y) {
+  return x ^ y;
+}
+v128 TESTFN v128_bitselect(v128 x, v128 y, v128 cond) {
+  return (v128)__builtin_wasm_bitselect((i32x4)x, (i32x4)y, (i32x4)cond);
+}
+i8x16 TESTFN i8x16_neg(i8x16 vec) {
+  return -vec;
+}
+int32_t TESTFN i8x16_any_true(i8x16 vec) {
+  return __builtin_wasm_any_true_i8x16(vec);
+}
+int32_t TESTFN i8x16_all_true(i8x16 vec) {
+  return __builtin_wasm_all_true_i8x16(vec);
+}
+// i8x16 TESTFN i8x16_shl(i8x16 vec, int32_t shift) {
+//   return vec << shift;
+// }
+// i8x16 TESTFN i8x16_shr_s(i8x16 vec, int32_t shift) {
+//   return vec >> shift;
+// }
+// i8x16 TESTFN i8x16_shr_u(i8x16 vec, int32_t shift) {
+//   return (i8x16)((u8x16)vec >> shift);
+// }
```

Comment:
Toolchain and v8 disagree about whether the shift amount should be an immediate or stack value.

Reply:
Will prototype as stack values in V8, and report back on the open issue.